### PR TITLE
Add job board for global partners

### PR DESCRIPTION
**Pull request by amakarudze**
Every year, we have global partners and many companies reaching out to us asking us to post their job openings to our network. We are adding a job board as a way of upselling our sponsors and ask them to pay a small fee to advertise their job openings on our website.

I have added some tests, which may seem unnecessary, but it was necessary to make sure that the job board was implemented in such a way that only superusers can manage jobs and companies so we can control whose job adverts are on our site.

### CODE REVIEW

File: jobboard/admin.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from .models import Company, Job
+
+
+admin.site.register(Company)
+
+
+class JobAdmin(admin.ModelAdmin):
+    list_display = ('role', 'company', 'location', 'remuneration', 'open', 'date_created')
+    search_fields = ('company', 'role')
+    list_filter = ('open', 'company')
```
Comment (marksweb) [inline jobboard/admin.py:12]: I feel like filtering on `location` and `date_created` might be useful?

File: jobboard/migrations/0004_auto_20221220_2325.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 3.2.16 on 2022-12-20 23:25
```
Comment (marksweb) [inline jobboard/migrations/0004_auto_20221220_2325.py:1]: I'd delete all the migrations and create the initial again - that way this goes out with just 1 migration. Bit cleaner that way.

File: jobboard/views.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+from .models import Job
+
+
+def index(request):
+    job_list = Job.objects.filter(open=True).order_by("-date_created")
+    return render(request, 'jobboard/index.html', {'job_list': job_list})
+
+
+def job_detail(request, id):
+    job = Job.objects.get(id=id)
```
Comment (marksweb) [inline jobboard/views.py:12]: Should probably wrap this in a `try/except DoesNotExist` with a 404 returned if the `get()` fails.

File: templates/jobboard/index.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                                {{ job.role }}</a></h4>
+                              <p>{{ job.location }}</p>
+                              {% if job.remuneration %}
+                                  <p>{{ job.remuneration }}</p>
+                              {% endif %}
+                              <p>{{ job.summary }}</p></div>
+                          <p></p>
+                          <div class="row">
+                              <div class="col-md-6 text-center">
+                                  <a href="{{ job.apply }}" class="cta">Apply here</a>
```
Comment (marksweb) [inline templates/jobboard/index.html:39]:
```suggestion
                                  <a href="{{ job.apply }}" class="cta">{% trans "Apply here" %}</a>
```

File: templates/jobboard/index.html [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                              </div>
+                              <div class="col-md-6 text-center">
+                                  <a href="{% url 'jobboard:job_detail' job.id %}" class="cta">More information</a>
```
Comment (marksweb) [inline templates/jobboard/index.html:42]:
```suggestion
                                  <a href="{% url 'jobboard:job_detail' job.id %}" class="cta">{% trans "More information" %}</a>
```

File: templates/jobboard/job.html [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            <img src="{{ job.company.logo.url }}">
+        </div>
+        <h4>{{ job.role }}</h4>
+        <p>{{ job.location }}</p>
+        <p>{{ job.remuneration }}</p>
+
+        <p>{{ job.description| safe }}</p>
+        <br>
+        <div class="text-center">
+            <a href="{{ job.apply }}" class="btn btn-primary">Apply here</a>
```
Comment (marksweb) [inline templates/jobboard/job.html:16]:
```suggestion
            <a href="{{ job.apply }}" class="btn btn-primary">{% trans "Apply here" %}</a>
```

File: jobboard/models.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
+from django.db import models
+
+from tinymce import models as tinymce_models
+
+
+class Company(models.Model):
+    name = models.CharField(max_length=200)
```
Comment (marksweb) [inline jobboard/models.py:7]: Would it be worth having a slug field on the company which would enable a view which filtered the jobs by company?

**Review by marksweb — CHANGES_REQUESTED** [review]

File: jobboard/models.py (hunk 7 above)
Comment (amakarudze) [reply jobboard/models.py:7]: Done!

File: templates/jobboard/job.html (hunk 6 above)
Comment (amakarudze) [reply templates/jobboard/job.html]: Done!

File: templates/jobboard/index.html (hunk 5 above)
Comment (amakarudze) [reply templates/jobboard/index.html]: Done!

File: jobboard/views.py (hunk 3 above)
Comment (amakarudze) [reply jobboard/views.py]: Done!

File: templates/jobboard/index.html [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+        {% if forloop.first %}<div class="row">{% endif %}
+            <div class="col-md-6">
+                <div class="panel">
+                    <div class="card mb-4">
+                        <div class="card-body">
+                            <h3 class="card-title text-center">
+                              <a href="{% url 'jobboard:job_detail' job.id %}">{{ job.company.name }}</a>
+                              {% if job.date_created|timesince > "20 hours" %}
+                              <span class="badge badge-dprimary">New</span>
```
Comment (marksweb) [inline templates/jobboard/index.html:25]:
```suggestion
                              <span class="badge badge-dprimary">{% trans "New" %}</span>
```

**Review by marksweb — CHANGES_REQUESTED** [review]
Sorry, spotted one more untranslated string but you should just be able to accept my suggestion 👍

**Pull request merged**